geoportal/c2cgeoportal_geoportal/lib/metrics.py

```diff
@@ -27,86 +27,94 @@
 
 import gc
 import sys
-from typing import Any
+from collections.abc import Generator
 
+import prometheus_client.core
+import prometheus_client.registry
 from c2cwsgiutils import broadcast
 from c2cwsgiutils.debug import get_size
-from c2cwsgiutils.metrics import Provider
 
 from c2cgeoportal_geoportal.lib.caching import MEMORY_CACHE_DICT
 from c2cgeoportal_geoportal.views.raster import Raster
 
 
-class MemoryCacheSizeProvider(Provider):
+class MemoryCacheSizeCollector(prometheus_client.registry.Collector):
     """Get the memory used by the cache."""
 
     def __init__(self, all_: bool = False):
-        super().__init__("pod_process_memory_cache_kb", "Used memory cache")
+        super().__init__()
         self.all = all_
 
-    def get_data(self) -> list[tuple[dict[str, Any], float]]:
+    def collect(self) -> Generator[prometheus_client.core.GaugeMetricFamily, None, None]:
+        gauge = prometheus_client.core.GaugeMetricFamily(
+            "c2cgeoportal_memory_cache_bytes",
+            "Memory used by cache",
+            labels=["pid", "hostname", "key"],
+            unit="bytes",
+        )
+
         elements = _get_memory_cache(all_=self.all)
         assert elements is not None
         result = []
         for elem in elements:
             if elem is not None:
-                for value in elem["values"]:
-                    value[0]["pid"] = str(elem["pid"])
-                    value[0]["hostname"] = elem["hostname"]
+                for key, value in elem["values"]:
+                    gauge.add_metric([str(elem["pid"]), str(elem["hostname"]), key], value)
                     result.append(value)
-        return result
+        yield gauge
 
 
 @broadcast.decorator(expect_answers=True, timeout=15)
-def _get_memory_cache(all_: bool) -> dict[str, list[tuple[dict[str, Any], float]]]:
-    values = (
-        [({"key": key}, get_size(value) / 1024) for key, value in list(MEMORY_CACHE_DICT.items())]
-        if all_
-        else []
-    )
-    values.append(({"key": "total"}, get_size(MEMORY_CACHE_DICT) / 1024))
+def _get_memory_cache(all_: bool) -> dict[str, list[tuple[str, float]]]:
+    values = [(key, get_size(value) / 1024) for key, value in list(MEMORY_CACHE_DICT.items())] if all_ else []
+    values.append(("total", get_size(MEMORY_CACHE_DICT)))
```

Comment:
missing `/ 1024`?

Reply:
No, previously we were using kbytes, but Prometheus prefers bytes :-)

Reply:
OK :+1: but then maybe the `/ 1024` on the previous line (69) should be removed? (sorry I edited my previous comment at the same time you were responding)